modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-partner-portal/osb-site-initializer-partner-portal-test/src/testFunctional/macros/PRMMDFRequest.macro

```diff
@@ -0,0 +1,191 @@
+			Click.clickAtNotVisible(
+				key_fieldName = "targetAudienceRoles",
+				key_optionValue = "${targetRolesList}",
+				locator1 = "PRMMDFRequest#CHECKBOX");
+		}
+	}
+
+	@summary = "Format the date that will be inputed on start and end date"
+	macro formatDate {
+		var nextDayDate = DateUtil.getFormattedDate("+1", "MMddyyyy");
+
+		if (isSet(endDate)) {
+			var nextDayDate = DateUtil.getFormattedDate("+2", "MMddyyyy");
+		}
+
+		return "${nextDayDate}";
+	}
+
+	@summary = "Add a new activity"
+	macro newActivity {
+		if (!(isSet(activityIndex))) {
+			var activityIndex = "0";
+		}
+
+		Click(
+			key_buttonText = "Add Activity",
+			locator1 = "PRMMDFRequest#ADD_BUTTON");
+
+		Type(
+			key_fieldName = "activities[${activityIndex}].name",
+			locator1 = "PRMMDFRequest#TEXT_FIELD",
+			value1 = "${activityName}");
+
+		Select(
+			key_fieldName = "activities[${activityIndex}].typeActivity",
+			locator1 = "PRMMDFRequest#SELECT",
+			value1 = "${typeOfActivity}");
+
+		Select(
+			key_fieldName = "activities[${activityIndex}].tactic",
+			locator1 = "PRMMDFRequest#SELECT",
+			value1 = "${tactic}");
+
+		var startDate = PRMMDFRequest.formatDate();
+
+		Type.sendKeys(
+			key_fieldName = "activities[${activityIndex}].startDate",
+			locator1 = "PRMMDFRequest#TEXT_FIELD",
+			value1 = "${startDate}");
+
+		var endDate = PRMMDFRequest.formatDate(endDate = "true");
+
+		Type.sendKeys(
+			key_fieldName = "activities[${activityIndex}].endDate",
+			locator1 = "PRMMDFRequest#TEXT_FIELD",
+			value1 = "${endDate}");
+
+		var budgetIndex = "0";
+
+		for (var bugdetValues : list "${budgetList}") {
+			var expenseOption = StringUtil.extractFirst("${bugdetValues}", "-");
+			var bugdetValue = StringUtil.extractLast("${bugdetValues}", "-");
+
+			Click(
+				key_buttonText = "Add Expense",
+				locator1 = "PRMMDFRequest#ADD_BUTTON");
+
+			Select(
+				key_fieldName = "activities[${activityIndex}].budgets[${budgetIndex}].expense",
+				locator1 = "PRMMDFRequest#SELECT",
+				value1 = "${expenseOption}");
+
+			Type(
+				key_fieldName = "activities[${activityIndex}].budgets[${budgetIndex}].cost",
+				locator1 = "PRMMDFRequest#TEXT_FIELD",
+				value1 = "${bugdetValue}");
+
+			var budgetIndex = ${budgetIndex} + 1;
+		}
+
+		Click.clickAtNotVisible(
+			key_fieldName = "activities[${activityIndex}].leadGenerated",
+			key_optionValue = "${leadList}",
+			locator1 = "PRMMDFRequest#CHECKBOX");
+
+		Type(
+			key_fieldName = "activities[${activityIndex}].targetOfLeads",
+			locator1 = "PRMMDFRequest#TEXT_FIELD",
+			value1 = "${targetOfLeads}");
+
+		for (var leadFollowUp : list "${leadFollowUpList}") {
+			Click.clickAtNotVisible(
+				key_fieldName = "activities[${activityIndex}].leadFollowUpStrategies",
+				key_optionValue = "${leadFollowUp}",
+				locator1 = "PRMMDFRequest#CHECKBOX");
+		}
+
+		Type(
+			key_fieldName = "activities[${activityIndex}].detailsLeadFollowUp",
+			locator1 = "PRMMDFRequest#TEXT_FIELD",
+			value1 = "${followUpStrategy}");
+	}
```

Comment:
This macro looks so big... is there a way to split it? Let me know if there isn't as well, I trust your judgement.

Reply:
I think that it's not possible. This section from PRM is a form with a lot of fields.